test/funnels/Depositor.t.sol

```diff
@@ -0,0 +1,595 @@
+        });
+        vm.prank(FACILITATOR); (uint128 liq, uint256 deposited0, uint256 deposited1) = depositor.deposit(dp);
+        assertGt(_getLiquidity(DAI, USDC, 100, REF_TICK-100, REF_TICK+100), 0);
+
+        dp.liquidity = 0;
+        dp.amt0Desired = deposited0;
+        dp.amt1Desired = deposited1;
+
+        uint256 liquidityBeforeWithdraw = _getLiquidity(DAI, USDC, 100, REF_TICK-100, REF_TICK+100);
+
+        vm.warp(block.timestamp + 3600);
+        vm.expectEmit(true, true, true, false);
+        emit Withdraw(FACILITATOR, DAI, USDC, liq, 0, 0, 0, 0);
+        vm.prank(FACILITATOR); (, uint256 withdrawn0, uint256 withdrawn1) = depositor.withdraw(dp, false);
+
+        // due to liquidity from amounts calculation there is rounding dust
+        assertTrue(withdrawn0 * 100001 / 100000 >= deposited0);
+        assertTrue(withdrawn1 * 100001 / 100000 >= deposited1);
+        assertEq(GemLike(DAI).balanceOf(address(depositor)), 0);
+        assertEq(GemLike(USDC).balanceOf(address(depositor)), 0);
+        assertLe(_getLiquidity(DAI, USDC, 100, REF_TICK-100, REF_TICK+100), liquidityBeforeWithdraw);
+    }
+
+    function testDepositWrongGemOrder() public {
+        Depositor.LiquidityParams memory dp = Depositor.LiquidityParams({
+            gem0: USDC,
+            gem1: DAI,
+            fee: uint24(100),
+            tickLower: REF_TICK-100,
+            tickUpper: REF_TICK+100,
+            liquidity: 0,
+            amt0Desired: 0,
+            amt1Desired: 0,
+            amt0Min: 0,
+            amt1Min: 0
+        });
+        vm.expectRevert("Depositor/wrong-gem-order");
+        vm.prank(FACILITATOR); depositor.deposit(dp);
+    }
+
+    function testDepositTooSoon() public {
+        Depositor.LiquidityParams memory dp = Depositor.LiquidityParams({
+            gem0: DAI,
+            gem1: USDC,
+            fee: uint24(100),
+            tickLower: REF_TICK-100,
+            tickUpper: REF_TICK+100,
+            liquidity: 0,
+            amt0Desired: 500 * WAD,
+            amt1Desired: 500 * 10**6,
+            amt0Min: 490 * WAD,
+            amt1Min: 490 * 10**6
+        });
+        vm.prank(FACILITATOR); depositor.deposit(dp);
+        vm.expectRevert("Depositor/too-soon");
+        vm.prank(FACILITATOR); depositor.deposit(dp);
+    }
+
+    function testDepositExceedingCap() public {
+        (,,uint128 cap0, uint128 cap1) = depositor.limits(DAI, USDC);
+        Depositor.LiquidityParams memory dp = Depositor.LiquidityParams({
+            gem0: DAI,
+            gem1: USDC,
+            fee: uint24(100),
+            tickLower: REF_TICK-100,
+            tickUpper: REF_TICK+100,
+            liquidity: 0,
+            amt0Desired: 2 * cap0,
+            amt1Desired: 2 * cap1,
+            amt0Min: 0,
+            amt1Min: 0
+        });
+
+        vm.expectRevert("Depositor/exceeds-cap");
+        vm.prank(FACILITATOR); depositor.deposit(dp);
+    }
+
+    function testDepositExceedingSlippage() public {
```

Comment:
Same here for each amount.

Reply:
Done in [f23d585](https://github.com/makerdao/dss-allocator/pull/2/commits/f23d585b25acdf5f855375d4d67f5ec37a07aee0)